site/search/SearchResults.tsx

```diff
@@ -0,0 +1,267 @@
+    PageType,
+} from "./searchTypes.js"
+
+class ChartResult extends React.Component<{
+    hit: ChartHit
+    queryCountries: Country[]
+}> {
+    @computed get entities() {
+        return pickEntitiesForChart(this.props.hit, this.props.queryCountries)
+    }
+
+    @computed get slug(): string {
+        const { hit } = this.props
+        const { entities } = this
+        if (!entities.length) return hit.slug
+        else
+            return setSelectedEntityNamesParam(
+                Url.fromURL(hit.slug).updateQueryParams({
+                    tab: "chart",
+                }),
+                entities
+            ).fullUrl
+    }
+
+    @computed get title() {
+        const { hit } = this.props
+        const { entities } = this
+        if (!entities.length) return hit.title
+        else return hit.title.trim() + `, ${entities.join(", ")}`
+    }
+
+    render() {
+        const { hit } = this.props
+        const { slug, title } = this
+
+        return (
+            <li className="ChartResult">
+                {/* <a href={`${BAKED_GRAPHER_URL}/${hit.slug}`} dangerouslySetInnerHTML={{__html: hit._highlightResult.title.value}}/> */}
+                <a href={`${BAKED_GRAPHER_URL}/${slug}`}>{title}</a>
+                {hit.variantName ? (
+                    <span className="variantName"> {hit.variantName}</span>
+                ) : undefined}
+                {hit._snippetResult ? (
+                    <p
+                        dangerouslySetInnerHTML={{
+                            __html: hit._snippetResult.subtitle.value,
+                        }}
+                    />
+                ) : undefined}
+            </li>
+        )
+    }
+}
+
+class PageResult extends React.Component<{ hit: PageHit }> {
+    /**
+     * We want to decide whether to show the excerpt or the content, in addition to the title.
+     * The excerpt is usually more useful, but there can be cases where the content is better:
+     * - There is no excerpt.
+     * - We have a good search match in the content, but not in the excerpt or the title.
+     */
+    @computed get textToShow(): string {
+        const { hit } = this.props
+        if (!hit._snippetResult?.excerpt?.value?.length)
+            return hit._snippetResult?.content?.value || ""
+
+        const highlightMatches = [
+            { name: "title", ...hit._highlightResult.title },
+            { name: "excerpt", ...hit._snippetResult.excerpt },
+            { name: "content", ...hit._snippetResult.content },
+        ]
+        const algoliaMatchLevels = ["none", "partial", "full"]
+        const highlighted = highlightMatches.map((e) => ({
+            ...e,
+            matchLevel: algoliaMatchLevels.indexOf(e.matchLevel ?? ""),
+        }))
+        const best = maxBy(highlighted, (h) => h.matchLevel)
+        if (best?.name === "content")
+            return hit._snippetResult.content?.value || ""
+        else return hit._snippetResult.excerpt?.value || ""
```

Comment:
Yes, it's actually as you say, and that was actually my intention in this case.
If the search term appears in the title already, then it's probably occurring tens of times in `content`, and some out-of-context snippet from the fulltext doesn't help much. my thinking there is that the excerpt is more helpful in that case, even if it doesn't contain the search term.

but I can also see how you might be thinking differently, and I can change it back to have it the way you described.

Reply:
one example is here: in this case, I would much rather see the concise excerpt, rather than some out-of-context occurence of the word "coronavirus" within the article.

![image](https://user-images.githubusercontent.com/2641501/217568881-8b85c533-96df-4244-8bae-bbc07e9743ce.png)

Reply:
> If the search term appears in the title already, then it's probably occurring tens of times in content, and some out-of-context snippet from the fulltext doesn't help much. my thinking there is that the excerpt is more helpful in that case, even if it doesn't contain the search term.

what you describe here makes sense to me, I'd stick with it. Thanks for explaining your thinking